mongo/integration/client_side_encryption_test.go

```diff
@@ -235,3 +237,146 @@ func TestClientSideEncryptionWithExplicitSessions(t *testing.T) {
 		assert.NotEqual(mt, lsid, session.ID(), "expected different lsid, but got %v", lsid)
 	})
 }
+
+// customCrypt is a test implementation of the driver.Crypt interface. It keeps track of the number of times its
+// methods have been called.
+type customCrypt struct {
+	numEncryptCalls              int
+	numDecryptCalls              int
+	numCreateDataKeyCalls        int
+	numEncryptExplicitCalls      int
+	numDecryptExplicitCalls      int
+	numCloseCalls                int
+	numBypassAutoEncryptionCalls int
+}
+
+var (
+	_     driver.Crypt = (*customCrypt)(nil)
+	mySSN              = "123456789"
+)
+
+// Encrypt encrypts the given command.
+func (c *customCrypt) Encrypt(ctx context.Context, db string, cmd bsoncore.Document) (bsoncore.Document, error) {
+	c.numEncryptCalls++
+	elems, err := cmd.Elements()
+	if err != nil {
+		return nil, err
+	}
+
+	encryptedCmd := bsoncore.NewDocumentBuilder()
+	for _, elem := range elems {
+		// "encrypt" ssn element as "hidden"
+		if elem.Key() == "ssn" {
+			encryptedCmd = encryptedCmd.AppendString("ssn", "hidden")
+		} else {
+			encryptedCmd = encryptedCmd.AppendValue(elem.Key(), elem.Value())
+		}
+	}
+	return encryptedCmd.Build(), nil
+}
+
+// Decrypt decrypts the given command response.
+func (c *customCrypt) Decrypt(ctx context.Context, cmdResponse bsoncore.Document) (bsoncore.Document, error) {
+	c.numDecryptCalls++
+	elems, err := cmdResponse.Elements()
+	if err != nil {
+		return nil, err
+	}
+
+	decryptedCmdResponse := bsoncore.NewDocumentBuilder()
+	for _, elem := range elems {
+		// "decrypt" ssn element as mySSN
+		if elem.Key() == "ssn" {
+			decryptedCmdResponse = decryptedCmdResponse.AppendString("ssn", mySSN)
+		} else {
+			decryptedCmdResponse = decryptedCmdResponse.AppendValue(elem.Key(), elem.Value())
+		}
+	}
+	return decryptedCmdResponse.Build(), nil
+}
+
+// CreateDataKey implements the driver.Crypt interface.
+func (c *customCrypt) CreateDataKey(ctx context.Context, kmsProvider string, opts *mcopts.DataKeyOptions) (bsoncore.Document, error) {
```

Comment:
Optional: When the `unparam` linter is enabled, it will complain about these unused parameter variables. Consider removing the variable names.

E.g.
```go
 func (c *customCrypt) CreateDataKey(context.Context, string, *mcopts.DataKeyOptions) (bsoncore.Document, error)
```

Consider the same change for all other functions on `*customCrypt`.

Reply:
Good catch; done. But, I had to use `_` as the names FYI. Removing the names entirely doesn't seem to be correct syntax.